### DISCUSSION TITLE
是否支持高亮某一个字符串(全匹配或者忽略大小写匹配)？

### DISCUSSION THREAD

**Discussion by Robin329** [ayamir/nvimdots#952] [Q&A] [answered]
### Feature description

类似vscode中的highlight words插件，可以高亮某一个字符串.
![image](https://github.com/ayamir/nvimdots/assets/22770951/7614a990-51ec-4cb5-992f-51851f518053)

**Comment by Jint-lzxy** [comment]
This should be in discussions, transferred.

**Comment by Jint-lzxy** [comment]
Maybe you're looking for `:h search-commands` `:h *` `:h #` Search commands or [RRethy/vim-illuminate](https://github.com/RRethy/vim-illuminate)?

**Comment by Robin329** [comment]
同时可以标记多个字符串，并且支持不同的颜色：
![image](https://github.com/ayamir/nvimdots/assets/22770951/9a9dada9-4ed4-4797-a4c6-dc13fae203fb)

对应vscode的配置为：
```json

    "highlightwords.colors": [
        {
            "dark": "yellow"
        },
        {
            "dark": "Cyan"
        },
        {
            "dark": "Pink"
        },
        {
            "dark": "LightGreen"
        },
        {
            "dark": "LightSteelBlue"
        },
        {
            "dark": "Plum"
        }
    ],
    "highlightwords.box": {
        "light": true,
        "dark": false
    },
    "highlightwords.showSidebar": true,


    "highlightwords.defaultMode": 1,
    "workbench.colorCustomizations": {
        "editor.selectionBackground": "#989b9b",
        "editorCursor.foreground": "#ffffff",
        "editorLineNumber.foreground": "#5e615f"
    },
```

**Reply by Jint-lzxy** [reply, marked as answer]
Maybe u're looking for [vim-mark](https://github.com/inkarkat/vim-mark) or its alternatives?

**Reply by Robin329** [reply]
是的，是否有这样的插件或者更好的推荐？

**Reply by Jint-lzxy** [reply]
It's okay to install vim plugins. If this meets your needs u may [add this new plugin](https://github.com/ayamir/nvimdots/wiki/Usage#add-a-new-plugin) to ur config.

**Comment by Robin329** [comment]
我添加该插件之后报如下错误，您知道是什么原因吗？
![image](https://github.com/ayamir/nvimdots/assets/22770951/9aa84bb3-cf5a-489f-819b-95c936a1ba43)

**Reply by Jint-lzxy** [reply]
See https://github.com/ayamir/nvimdots/issues/901. Setting up vim plugins is usually very straightforward - they only need to be "fed" some global options.

**Reply by Robin329** [reply]
我拿掉了setup()，还是报如下错误：
![image](https://github.com/ayamir/nvimdots/assets/22770951/0a740ff9-11da-4bc1-96df-391b7a612ff3)

新的配置如下：
```lua
return function()
    vim.g.mwDefaultHighlightingPalette = 'maximum'
    vim.g.mwDefaultHighlightingNum = 9
    vim.g.mwAutoLoadMarks = 1
    vim.g.mwAutoSaveMarks = 0
    vim.g.mwIgnoreCase = 0
end


```

**Reply by Jint-lzxy** [reply]
`<leader>r` is already defined here:
https://github.com/ayamir/nvimdots/blob/10e2e491402de420c38ebfa654536a2bd6ff1986/lua/keymap/tool.lua#L19-L21

Since `vim-mark` define `<leader>r` as:
```vim
if !hasmapto('<Plug>MarkRegex', 'n')
	nmap <unique> <Leader>r <Plug>MarkRegex
"            ^^^^^^^^
endif
```
You need to give it a new mapping. I think the author has already documented this in detail in [README.md](https://github.com/inkarkat/vim-mark/tree/master#configuration:~:text=If%20you%20want%20no%20or%20only%20a%20few%20of%20the%20available%20mappings%2C%20you%20can%20completely%20turn%20off%20the%20creation%20of%20the%20default%20mappings%20by%20defining%3A).

**Reply by Robin329** [reply]
通过设置该属性，应该会忽略到默认的按键映射，但是冲突的按键映射还是会有。
`vim.g.mw_no_mappings = 1`

所以如果 我不修改vim-mark的代码，是否可以重新进行映射？

**Reply by Jint-lzxy** [reply]
> 所以如果 我不修改vim-mark的代码，是否可以重新进行映射？


In this case, you need to define these mappings yourself. sth like:
```lua
["n|<some-keymap>"] = map_cmd("<Plug>MarkRegex"):with_noremap():with_silent(),
```

**Reply by Robin329** [reply]
![image](https://github.com/ayamir/nvimdots/assets/22770951/8ecfea4b-6dc3-4b46-aed3-7da0b712b8c3)
使用`<space>m`报以上错误，不知道是否是设置问题？

**Reply by Jint-lzxy** [reply]
> 使用`<space>m`报以上错误，不知道是否是设置问题？

Ys. I don't have your config, so you need to debug it yourself.
